autowiring/CoreContext.h

```diff
@@ -573,23 +573,35 @@ class CoreContext:
   /// Creation helper routine
   /// </summary>
   template<class T>
-  static std::shared_ptr<CoreContext> Create(
+  static std::shared_ptr<CoreContextT<T>> Create(
     std::shared_ptr<CoreContext> pParent,
     t_childList::iterator backReference
   ) {
-    return std::static_pointer_cast<CoreContext>(
-      std::make_shared<CoreContextT<T>>(pParent, backReference)
-    );
+    return std::make_shared<CoreContextT<T>>(pParent, backReference);
+  }
+
+private:
+  /// \internal
+  /// <summary>
+  /// Creation helper routine
+  /// </summary>
+  template<class T>
+  static std::shared_ptr<CoreContext> UntypedCreate(
```

Comment:
Prefer the name `CreateUntyped` instead.  This causes all `Create` methods to be near each other in sorted statement completion lists.